platform/wayland/cog-platform-wl.c

```diff
@@ -1173,6 +1239,14 @@ registry_on_global_remove(void *data, struct wl_registry *registry, uint32_t nam
             break;
         }
     }
+
+    CogWlSeat *seat, *tmp_seat;
+    wl_list_for_each_safe(seat, tmp_seat, &platform->display->seats, link) {
+        if (seat->seat_name == name) {
+            cog_wl_seat_destroy(seat);
```

Comment:
If theIf the seat being destroyed is the current one, we should check that and set it to `NULL`.

Reply:
good point

Reply:
PR updated.